deployments/rbac/clusterrole.yaml

```diff
@@ -6,5 +6,5 @@ metadata:
 rules:
   - apiGroups:
         - ""
-    resources: ["services", "nodes", "configmaps"]
+    resources: ["services", "nodes", "configmaps", "secrets"]
```

Comment:
A [quick ChatGPT session](https://chat.openai.com/share/8f8f8c88-4116-4579-84f7-583b65862355) suggests access to Secrets is reasonable for a controller. Also, limiting the access to the controller's namespace is definitely a good practice.

One problem with making this optional when using the `no-tls` mode is that the TLS Mode can be configured on the fly without requiring a deployment. To change the RBAC to remove that permission would require another deployment. Additionally, if `no-tls` is being used presumably there are no Secrets in the `nlk` namespace so no danger of anything being leaked.

Reply:
Sounds like good due diligence. I think since this works in a namespace and the secrets exist there it should not be controversial.  I'm not well versed in Kubernetes security practices but we can let the community tell us if this is a bad call.